WIP(web-diary): npm package content

gochicus opened the pull request:
ISSUES CLOSED: #81

gochicus committed:
WIP(web-diary): npm package content

ISSUES CLOSED: #81

TFK70 commented:
@gochicus когда ты создаешь папку - подразумевается, что там будет лежать больше одного элемента (иначе зачем папка)
Это я к тому что в том что ты засунул свой пакет в папку package не особо много смысла, переименуй ее в packageS

TFK70 commented on package/list/package.json:
```diff
@@ -0,0 +1,25 @@
+  "description": "item iteration",
+  "main": "index.ts",
+  "scripts": {
+    "test": "echo \"Error: no test specified\""
```
Это что?

TFK70 commented on package/list/package.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "@gochicus/list",
```
А почему @academy/gochicus не получилось?

gochicus replied:
@TFK70 Я  не совсем уверен, долго искал эту проблему и наткнулся на вот  [такое](https://rugeek.ru/questions/npm-403-forbidden-put-http-registry-npmjs-org-package-name-forbidden/)
К слову, ты смотрел  type ошибку с `Text`, про которую я в issue писал?

TFK70 replied:
> К слову, ты смотрел type ошибку с Text, про которую я в issue писал?

В следующий раз оставляй пожалуйста ссылку
Там импорта нет

> Я не совсем уверен, долго искал эту проблему и наткнулся на вот такое

И ты зачем-то оттуда скопипстил шаги? Есть идеи почему не сработало-то?

gochicus replied:
@TFK70 

> И ты зачем-то оттуда скопипстил шаги? Есть идеи почему не сработало-то?

Там нечего копипастить, но написано, что в скоупе должно быть имя юзера, совпадающее с тем, под которым ты зарегистрирован на npm. Я в скоуп ник и записал. А почему скоуп academy не срабатывает, ну думаю от того, что довольно распространенное название.  К слову, как я уже писал, под скоупом gochicus/list пакет нормально публикуется.

TFK70 commented on package/list/src/Item.tsx:
```diff
@@ -0,0 +1,86 @@
+import React, { useState }               from 'react'
+
+import { Button }                         from './button/src'
```
Что-то не так с путями
src создается в случае с пакетом, сейчас твои button'ы и condition'ы пакетами не являются (по-крайней мере не должны ими являться), так что если у тебя снова структура button->src где на одном уровне с src ничего не лежит то вынеси все на уровень выше

TFK70 commented on package/list/src/button/src/Button.tsx:
```diff
@@ -0,0 +1,7 @@
+import styled                                  from '@emotion/styled'
```
Если ты называешь файлы по аналогии с uikit'ом - этот файл будет `button.component.tsx`

TFK70 commented on package/list/src/layout/src/Box.tsx:
```diff
@@ -0,0 +1,31 @@
+  color,
+  flexbox,
+  shadow,
+  position,
```
Неужели в листе прям все надо?

TFK70 commented on package/list/src/layout/src/Layout.tsx:
```diff
@@ -0,0 +1,8 @@
+import styled                                            from '@emotion/styled'
+import { color, flexbox, layout, position, size, space } from 'styled-system'
+
+export const Layout = styled.div(layout, space, flexbox, position, color, size)
```
И тут

gochicus committed:
WIP(web-diary): bpm edited content

affects: @practice/web-diary

ISSUES CLOSED: #81

gochicus committed:
feat(web-diary): npm package used in pist-list fragment

affects: @fragments/post-list, @practice/web-diary

ISSUES CLOSED: #81

TFK70 commented on practice/web-diary/fragments/post-list/src/PostList.tsx:
```diff
@@ -1,7 +1,7 @@
 import React                                               from 'react'
+import { List }                                            from '@gochicus/list/src'
```
а почему не @ui/layout/src @ui/button/src @store/post-data/src ?

TFK70 replied:
@gochicus и я не пойму куда сам пакет делся

gochicus replied:
@TFK70 без src почему-то модуль не находит

> @gochicus и я не пойму куда сам пакет делся

Он прямо из npm ставится, зачем он здесь?

TFK70 replied:
> @TFK70 без src почему-то модуль не находит

Вот давай ты в таких моментах не будешь сидеть над проблемой несколько часов а потом забивать на нее, а сразу обращаться ко мне? У тебя такое уже который раз повторяется

TFK70 replied:
> @TFK70 без src почему-то модуль не находит
> 
> > @gochicus и я не пойму куда сам пакет делся
> 
> Он прямо из npm ставится, зачем он здесь?

Он действительно берется из npm регистра, но это не значит что нужно убивать его локальную копию
То что ты его запаблил в глобальный регистр значит что ты сможешь использовать его где угодно за пределами репозитория где он живет, но как ты будешь в случае чего обновлять сам пакет?

gochicus replied:
@TFK70 я над ней и не сидел, оно не работало, добавил src - заработало, если в этом где-то проблема, то обсудим. Я лично хз почему так происходит,.  Насчет содержимого пакета, надо всё же реально на это смотреть, по завершении тудушки я с ней больше ничего и никогда делать не буду, в том числе пакет обновлять.

TFK70 replied:
@gochicus так у тебя щас к исходникам пакета есть доступ?

gochicus replied:
@TFK70 есть, а если бы и не было, качаешь из npm,идешь в node modules, оттуда его тащишь и меняешь как вздумается для новой версии.

TFK70 replied:
> @TFK70 есть, а если бы и не было, качаешь из npm,идешь в node modules, оттуда его тащишь и меняешь как вздумается для новой версии.

Ну насчет удобства такой "методики" я лучше промолчу, но то что ты понимаешь что такое нод модули это очень хорошо, вот только в них у тебя должен находиться только js. Что там делать исходникам?

gochicus replied:
@TFK70 почему только js? Как ты вообще это видишь? Сейчас модуль крайне прост: он просто дает подтянуть кусок реакта в код, а затем всё это компилится  и парсится вебпаком.

TFK70 replied:
@gochicus https://coderoad.ru/30928253/Написание-npm-модулей-в-TypeScript

TFK70 replied:
@gochicus нужно пояснение? Или все понятно?

gochicus replied:
@TFK70  а какой ответ мне нужен? Имеется ввиду `.npmignore`.?Что-то не пойму что к чему. Как сюда вписывается js?

gochicus replied:
@TFK70 Короче, я так и не понял как оно должно выглядеть и работать.

TFK70 replied:
@gochicus чекни еще связанный топик: https://coderoad.ru/13645824/Могу-ли-я-написать-пакет-npm-в-CoffeeScript

Там про coffee script но суть та же: когда ты захочешь юзать этот пакет с чистым js он будет орать на любой непонятный ему синтаксис, будь то jsx или определение типов через тайпскрипт, так что все твои супер-современные либы на последнем синтаксисе должны так или иначе полифилиться в js, чтобы их можно было использовать в любой другой среде

Плюс это weight issue, src как правило жирные, в отличие от сбилженных файлов, да они и не нужны в нод модулях никому

Чекни в тех же нод модулях как выглядит реакт, обрати внимание на его package.json

gochicus replied:
@TFK70 ты хочешь, чтобы модуль выдавал javascript и уже в код вставлял javascript или что?

TFK70 replied:
А почему "я хочу"? И что кого куда вставляет? Это просто нормы которые нужно соблюдать. Ну и внятного ответа на вопрос "зачем в нод модулях исходники" ты мне так и не дал

gochicus replied:
@TFK70 

> Ну и внятного ответа на вопрос "зачем в нод модулях исходники" ты мне так и не дал

Потому что ты внятно  такого вопроса не задавал. Но если на то пошло, у меня в пакете все файлы импортируюся локально, затем объединяются в лист, а он уже экспортируется. 

>  И что кого куда вставляет? Это просто нормы которые нужно соблюдать. 

Мы сейчас не абстракцию обсуждаем, а в контексте конкретной тудушки. У меня есть фрагмент с реактом, который компилится, затем парсится. Часть этого фрагмента приходит из npm пакета. Что-то мне подсказывает, что если я в реакт насыплю чистого javascript, webpack не обработает всё, радуясь,  что соблюдаются нормы. Фрагмент у нас не распарсится, приложение отображаться не будет.  И тут мы подходим к самому важному вопросу: что делать с javascript из модуля?  Я тебя в начале беседы спросил [как ты это видишь](https://github.com/atls-academy/gochicus.github.io/pull/87#discussion_r659535246), потому что, на мой взгляд, приложение сломается на этапе компиляции.  Как этот полученный javascript должен взаимодействовать с приложением?

TFK70 replied:
> Что там делать исходникам?

Давай без этого пожалуйста, если я задаю вопрос - значит в большинстве случаев ожидаю ответ. Ответ я не получил. (https://github.com/atls-academy/gochicus.github.io/pull/87#discussion_r659531919)

> Но если на то пошло, у меня в пакете все файлы импортируюся локально, затем объединяются в лист, а он уже экспортируется.

? Ты ведешь к тому что если тайпскрипт сбилдится в js то он сломается? Или в к тому что cjs модули не поддерживает?

> Что-то мне подсказывает

Дашь ссылку? Покажешь пример? Или это "что-то" не материальное?

> на мой взгляд

Твой взгляд формируется из предыдущего "что-то"?

Короче это все к тому что на данный момент в твоем описании слишком мало объективщины. Если ты утверждаешь что все сломается - либо подробно распиши какой момент и почему сломается (с ссылками на реальные сущности, а не на "что-то", например "webpack в этом моменте не отработает корректно потому что ..."), либо просто предоставь пример

К слову все этом время ты по-прежнему мог прикладывать код к своим комментам, он бы точно не помешал. Мы уже больше часа переписываемся текстом, а я так и не понял в чем твоя проблема, учитывая что я в целом знаю что в твоем приложении происходит. 

> Я тебя в начале беседы спросил как ты это видишь

И я по-прежнему не собираюсь писать за тебя код

gochicus replied:
@TFK70 

> ? Ты ведешь к тому что если тайпскрипт сбилдится в js то он сломается? Или в к тому что cjs модули не поддерживает?

 Вот фрагмент  из post-list:

>  return (
    <Row justifyContent='center'>
      <List
        deleteItem={deleteItem}
        likeItem={likeItem}
        importantItem={importantItem}
        items={filterData(postData, status, searchResult)}
        setPostData={setPostData}
      />
    </Row>

Меня волнует то, что  если вместо List там будет javascript не сломается ли приложение, когда webpack  будет парсить  react.  
Возвращаемся к этому вопросу:

> ты хочешь, чтобы модуль выдавал javascript и уже в код вставлял javascript или что?

То есть, вариантов два, либо "да", всё будет ок, либо "нет".  Если нет, то меня интересовал вопрос как это должно 
 правильно взаимодействовать в приложении, но ты считаешь последнее написанием кода за меня.

TFK70 replied:
@gochicus если что, jsx это просто более удобный синтаксис. Когда ты пишешь

```javascript
<List
   items={[1,2,3]}
/>
```

Под капотом происходит что-то такое

```javascript
React.createElement(List, { items: [1,2,3] })
```

Все, что ты пишешь - это в конечном итоге js. Браузеры с нодой не умеют исполнять ts, этот язык компилируемый. После компиляции он превращается в js

Ну и попробуй сам уже ответить на свой вопрос: если в браузере в среде js'а выполнится js - что-то сломается? И ты за это время мог уже тысячу раз это опробовать и скинуть лог с ошибкой при наличии

> То есть, вариантов два, либо "да", всё будет ок, либо "нет".

Не особо много от меня смысла, если б я так отвечал. Твоих знаний более чем достаточно чтобы самому отвечать на такие вопросы, опять же - посмотри как сделаны другие либы, если не хватает примеров. Реакт, наноид, либы атлантиса. Никто же не запрещает пробовать, в конце концов

gochicus committed:
refactor(web-diary): new package version used

affects: @gochicus/list, @ex-ui/background, @ex-ui/button, @ex-ui/input, @ex-ui/layout, @ex-ui/text,
@ex-ui/theme, @fragments/post-list, @practice/web-diary, @ui/background, @ui/button, @ui/input,
@ui/layout, @ui/text, @ui/theme

package code in list directory

ISSUES CLOSED: #81

gochicus committed:
style(list): lint changes

affects: @gochicus/list

ISSUES CLOSED: #81

TFK70 commented on package.json:
```diff
@@ -21,6 +21,7 @@
     "lessons/**/*",
-    "practice/**/*"
+    "practice/**/*",
+    "list/**"
```
Это единственная причина по которой у тебя все работает
В нод модулях у пакета нету папки dist, в поле main указан путь до несуществующего файла

TFK70 replied:
На всякий случай: с самим пакетом у тебя все ок (вроде как), просто сейчас он берется локально как воркспейс, соответственно в нод модули тянутся src

gochicus replied:
@TFK70 Нет, всё прекрасно работает и без воркспейсов, в list только исходники без дистов потому, что я js файлы почистил. Попробуй установи из npm и удали list из воркспейсов, у меня всё работает.

gochicus replied:
@TFK70 Нет, всё прекрасно работает и без воркспейсов, в list только исходники без дистов потому, что я js файлы почистил. Попробуй установи из npm и удали list из воркспейсов, у меня всё работает.

TFK70 replied:
> @TFK70 Нет, всё прекрасно работает и без воркспейсов, в list только исходники без дистов потому, что я js файлы почистил. Попробуй установи из npm и удали list из воркспейсов, у меня всё работает.

Ну так я про это и написал, что с пакетом все в порядке, убери его из воркспейсов

TFK70 commented on list/src/icons/svg/star.svg:
```diff
@@ -0,0 +1,37 @@
+</g>
+<g>
+</g>
+<g>
```
Почисти

gochicus replied:
@TFK70  У меня, кажется, проблемы, с линтом,  изменения после линта, добавленные в гит, не уходят в коммитах.

TFK70 replied:
Если git status их подсвечивает попробуй сделать коммит просто через git commit

TFK70 commented on list/src/icons/svg/heart.svg:
```diff
@@ -0,0 +1,40 @@
+<g>
+</g>
+<g>
+</g>
```
И тут

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,8 @@
+  importantItem: [{}]
+  likeItem: [{}]
+  postObject: {}
+  setPostData: [{}]
```
Насколько мне известно это коллбек

gochicus committed:
style(list): lint changes

affects: @gochicus/list

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,8 @@
+  deleteItem: [{}]
+  importantItem: [{}]
+  likeItem: [{}]
+  postObject: {}
```
Мало того что это объект. Какие проперти там должны быть?

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,8 @@
+export interface ListInterface {
+  items: [{ id: number }]
+  deleteItem: [{}]
```
Это вроде тоже коллбек

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,8 @@
+export interface ListInterface {
+  items: [{ id: number }]
```
А тут если не ошибаюсь свойств чуть больше, чем id

TFK70 commented on list/src/List.tsx:
```diff
@@ -0,0 +1,30 @@
+  importantItem,
+  likeItem,
+  setPostData,
+}: ListInterface): ReactJSXElement => {
```
Лучше вместо ReactJSXElement и тд писать:

```javascript
import { FC } from 'react'

const List: FC<ListProps> = () => ....
```

Более ёмко и лаконично

gochicus committed:
style(list): lint changes

ISSUES CLOSED: #81

gochicus committed:
style(list): lint changes

affects: @gochicus/list

ISSUES CLOSED: #81

gochicus committed:
style(list): lint changes

affects: @ex-ui/background, @ex-ui/button, @ex-ui/input, @ex-ui/layout, @ex-ui/text, @ex-ui/theme,
@practice/web-diary, @ui/background, @ui/button, @ui/input, @ui/layout, @ui/text, @ui/theme

ISSUES CLOSED: #81

gochicus committed:
refactor(list): types changed

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,8 +1,8 @@
-  postObject: {}
-  setPostData: [{}]
+  items: [{ post?: string; important?: boolean; like?: boolean; id: number }]
+  deleteItem: []
```
Еще раз:

1. Это коллбек
2. Даже если бы это был массив - указывай из каких элементов он состоит. Сейчас исходя из твоего интерфейса я могу написать в пропсах deleteItem={[1]} и все будет прекрасно работать

gochicus replied:
@TFK70 никогда не слышал про коллбек с использованием квадратных скобок. Есть ссылки, где об этом сказано? Лично я ориентируюсь на вот  [это](http://typescript-lang.ru/docs/Basic%20Types.html#Tuple)
upd Позже понял что ты имеешь в виду.

TFK70 commented on list/package.json:
```diff
@@ -0,0 +1,28 @@
+    "build": "tsc"
+  },
+  "dependencies": {
+    "@atlantis-lab/tsconfig": "^0.1.12",
```
Почему это в обычных зависимостях?

gochicus committed:
refactor(list): types changed

ISSUES CLOSED: #81

gochicus committed:
refactor(list): types changed

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,8 +1,19 @@
+  deleteItem: (
+    postData: [{ post?: string; important?: boolean; like?: boolean; id: number }],
+    id: number,
+  ) => []
```
эти функции ничего не возвращают
ты же понимаешь зачем ты это типизируешь?

gochicus replied:
@TFK70 

>  ты же понимаешь зачем ты это типизируешь?

В данном случае, нет. 

> эти функции ничего не возвращают

Ну, по идее массив. Если я перепишу на Array<any> будет лучше?

TFK70 replied:
> В данном случае, нет.

Так с этого и надо было начать. Это нужно исключительно для того чтобы понимать что компонент ожидает на вход. Если ты определишь компонент так

```javascript
<List
   deleteItem={['wesfaew']}
   setPostData={123}
```

И он ни на что не поругается - это правильное поведение или нет?

gochicus committed:
refactor(list): interface changed

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,19 +1,19 @@
+    postData: [{ post?: string | number; important?: boolean; like?: boolean; id: number }],
     id: number,
-  ) => []
+  ) => Array<any>
```
Что изменилось?

TFK70 replied:
@gochicus короче я веду к тому чтобы ты типизировал item:

```javascript
interface Item { .... }
```

А потом использовал его для типизации коллбеков:

```javascript
interface ListProps {
   items: Item[]
   deleteItem: () => Item[]
   ....
}
```

gochicus replied:
@TFK70 так, я это только после коммита увидел, инет отваливался.  Но как-то я с айтемом  идею не понял, ведь это просто большой JSX компонент и не представляю как его типизация поможет, ведь он не хранит логики и данных. Не представляю как это поможет.

TFK70 replied:
@gochicus ты щас тупо дублируешь [{ post?: string | number; important?: boolean; like?: boolean; id: number }]
Вынеси это в type и используй

gochicus committed:
refactor(list): interface changed

affects: @fragments/post-list, @practice/web-diary

ISSUES CLOSED: #81

TFK70 commented on list/package.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "@gochicus/list2",
```
Почему 2?

gochicus replied:
@TFK70 для теста использовал другое название, чтобы не путать с npm.  Поменять обратно?

TFK70 replied:
Ну да

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,19 @@
+  deleteItem: (
+    postData: [{ post?: string | number; important?: boolean; like?: boolean; id: number }],
+    id: number,
+  ) => [{ post?: string | number; important?: boolean; like?: boolean; id: number }]
```
Я же показал пример: вынеси в отдельный интерфейс, зачем код дублировать?

gochicus replied:
@TFK70 если два файла, то папка types нужна будет?

TFK70 replied:
@gochicus а че зачем два файла?
Создай в этом же

gochicus committed:
refactor(list): package name changed

ISSUES CLOSED: #81

gochicus committed:
refactor(list): interface structure changed

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,19 +1,20 @@
+}
+
+export interface PostDataObject {
+  object: { post?: string | number; important?: boolean; like?: boolean; id: number }
```
Зачем?

TFK70 replied:
Зачем? x2

TFK70 replied:
Или PostDataObject['object'] удобнее чем PostDataObject?

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,19 +1,20 @@
+  deleteItem: (postData: [PostDataObject['object']], id: number) => [PostDataObject['object']]
   importantItem: (
-    postData: [{ post?: string | number; important?: boolean; like?: boolean; id: number }],
+    postData: [PostDataObject['object']],
```
Короче:

```javascript
export interface ListInterface {
  items: PostDataObject[]
  deleteItem: (postData: PostDataObject[], id: number) => PostDataObject[]
  importantItem: (
    postData: PostDataObject[],
    id: number,
    status: boolean,
  ) => PostDataObject[]
  likeItem: (
    postData: PostDataObject[],
    id: number,
    status: boolean,
  ) => PostDataObject[]
  postObject: PostDataObject
  setPostData: () => PostDataObject[]
}

export interface PostDataObject { post?: string | number; important?: boolean; like?: boolean; id: number }
```

Тут все понятно?

gochicus replied:
@TFK70 Да, тут надо было возвращать массивы с объектами типов PostDataObject, а я ошибся, не поставил скобки и возвращал просто объект PostDataObject.

TFK70 replied:
@gochicus да ты еще и дополнительную вложенность в интерфейсе зачем-то добавил

TFK70 replied:
@gochicus да и вообще: у тебя же типы у коллбеков одинаковые, вынеси их тоже

gochicus committed:
refactor(list): interface changed

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,20 +1,17 @@
-  ) => [PostDataObject['object']]
+  deleteItem: (CallbackInterface) => PostDataObject['object'][]
+  importantItem: (CallbackInterface) => PostDataObject['object'][]
+  likeItem: (CallbackInterface) => PostDataObject['object'][]
```
Так они же у тебя по-прежнему все 3 абсолютно идентичны, создай тип для коллбека и только его и используй

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,20 +1,17 @@
+  setPostData: () => PostDataObject['object'][]
 }
 
 export interface PostDataObject {
```
Ну и на будущее: имей привычку определять использующиеся в дальнейшем коде переменные до их непосредственного использования (перенеси вспомогательные интерфейсы наверх)

TFK70 replied:
И наверх ты ничего так и не вынес

gochicus committed:
refactor(list): interface change

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -1,17 +1,19 @@
+  items: [PostDataObject]
+  deleteItem: CallbackInterface['callback']
+  importantItem: CallbackInterface['callback']
+  likeItem: CallbackInterface['callback']
```
Я не понимаю зачем ты это делаешь
Ты только что исправил это в PostDataObject и сделал абсолютно аналогичное тут? Тебе что-то мешает просто type использовать?

```javascript
type Callback = ({ ... }) => PostDataObject[]
```

Интерфейсы типа предназначены для типизации классов, а классы - это объекты, из этого следует что любой интерфейс по дефолту как бы тоже объект, поэтому их удобно использовать для типизации пропсов (потому что пропсы - это объект). Если тебе как сейчас никакой объект типизировать не надо - просто используй type и почитай доку

gochicus replied:
@TFK70 

>  И наверх ты ничего так и не вынес

ЧТо ты имеешь ввиду? в смысле в файле?

TFK70 replied:
Ну да, у тебя вспомогательные интерфейсы определяются после того, как они используются, поэтому вынеси их наверх
Это просто ради кодстайла, код все-таки сверху вниз читают

gochicus committed:
refactor(list): interface change

ISSUES CLOSED: #81

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,17 @@
+  deleteItem: Callback
+  importantItem: Callback
+  likeItem: Callback
+  postObject: PostDataObject
```
Это что?

gochicus replied:
@TFK70 Тут всё верно, именно объект с указанными типами. То же самое, что и одиночный item, но названия разные для лучшего понимания кода.

TFK70 replied:
Он тут зачем? Я вот посмотрел в твой фрагмент и там такого пропа нет, да и я не совсем понимаю его смысл

```javascript
      <List
        deleteItem={deleteItem}
        likeItem={likeItem}
        importantItem={importantItem}
        items={filterData(postData, status, searchResult)}
        setPostData={setPostData}
      />
```

gochicus replied:
@TFK70 прокидывается в Item, используется для управления внешним видом в зависимости от свойств конкретного объекта. Во фрагменте он не нужен, всё управление происходит внутри Item`a

TFK70 replied:
@gochicus никуда он не прокидывается, ты интерфейсы перепутал
Item у тебя вообще не типизирован

gochicus replied:
@TFK70 Абсолютно не согласен с твоей оценкой. Но возникает ощущение, что мы разные вещи обсуждаем. Предлагаю созвониться, как у тебя время будет, и детально всё обсудить.

TFK70 commented on list/src/List.interface.ts:
```diff
@@ -0,0 +1,17 @@
+export type Callback = (id: number, postData?: PostDataObject, status?: boolean) => PostDataObject[]
+
+export interface ListInterface {
+  items: [PostDataObject]
```
Интересный тип
Знаешь что такое tuple? И в чем разница между PostDataObject[]?

gochicus replied:
@TFK70  кортеж, когда данные внутри разных типов.  Исправлю на это PostDataObject[].

TFK70 replied:
@gochicus тут скорее суть была в том что их ограниченное количество, но да, с помощью таплов так же можно типизировать массив где разные типы

gochicus committed:
refactor(list): interface change

ISSUES CLOSED: #81

gochicus committed:
refactor(list): interface changed

ISSUES CLOSED: #81

TFK70 commented on list/src/Item.tsx:
```diff
@@ -1,10 +1,10 @@
-import { Text }                           from '@ui/text'
+import { Button }                         from './button'
+import { Condition }                      from './condition'
+import { HeartIcon, StarIcon, TrashIcon } from './icons/src'
```
Зачем они в src?

TFK70 replied:
@gochicus

TFK70 commented on list/src/List.tsx:
```diff
@@ -1,10 +1,17 @@
 
 import { Item as ListItem } from './Item'
+import { Column }           from './layout'
+import { ListInterface }    from './List.interface'
```
Если уж называешь в таком стиле:

`list.interface.ts`
`list.component.tsx`

TFK70 commented on list/src/icons/svg/heart.svg:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
И не совсем понимаю зачем тебе держать свг

TFK70 commented on practice/web-diary/webpack.config.ts:
```diff
@@ -23,7 +23,6 @@ export const module = {
   rules: [
     {
       test: /\.(ts|js)x?$/,
-      exclude: /node_modules/,
```
почему убрал?

gochicus committed:
refactor(list): file name changed

affects: @practice/web-diary

ISSUES CLOSED: #81

TFK70 commented on list/src/List.component.tsx:
```diff
@@ -5,7 +5,7 @@ import { Item as ListItem } from './Item'
 import { ListInterface }    from './List.interface'
 
-export const List: FC<ListInterface> = ({
+export const ListComponent: FC<ListInterface> = ({
```
Название-то List по-прежнему

gochicus committed:
refactor(list): component name change

ISSUES CLOSED: #81

gochicus committed:
refactor(list): icons moved from src

ISSUES CLOSED: #81

TFK70 commented on list/src/index.ts:
```diff
@@ -1 +1 @@
-export { List } from './List'
+export { ListComponent } from './List.component'
```
Ну так `list.component.tsx` и `List.component.tsx`, первая буква в лоуеркейсе

gochicus replied:
@TFK70 должна быть в лоуеркейсе?

TFK70 replied:
да

gochicus committed:
refactor(list): file name change

ISSUES CLOSED: #81

TFK70 commented on list/src/list.component.tsx:
```diff
@@ -3,7 +3,7 @@ import { FC }               from 'react'
 import { Item as ListItem } from './Item'
 import { Column }           from './layout'
-import { ListInterface }    from './List.interface'
+import { ListInterface }    from './list.interface'
```
Да, и лучше называть интерфейсы исходя из того что они типизируют, чтобы меньше путаницы было: `ListProps`

gochicus committed:
refactor(list): inteerface name change

ISSUES CLOSED: #81

gochicus committed:
refactor(list): names to lowercase

ISSUES CLOSED: #81

gochicus committed:
refactor(list): names to lowercase

ISSUES CLOSED: #81

gochicus committed:
refactor(list): files names changed

ISSUES CLOSED: #81

TorinAsakura commented on list/package.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "@gochicus/list",
+  "version": "0.0.14",
+  "description": "item iteration",
```
Что такое item iteration и что оно делает в описании пакета?

gochicus replied:
@TorinAsakura это, собственно, описание пакета.

TorinAsakura replied:
и как твоё описание даёт понимание о содержимом пакета? Либо сделай нормальное описание либо убери вообще это поле

TorinAsakura commented on list/src/button/base/base.styles.ts:
```diff
@@ -0,0 +1,12 @@
+
+export const createBaseStyles = (): styleFn => () => ({
+  cursor: 'pointer',
+  display: 'flex',
```
я полагаю, что это всё же потоковый объект, а не тупо блочный

gochicus replied:
@TorinAsakura можешь развернуть мысль? в чём тут проблема?

TorinAsakura replied:
в том, что это должно быть inline-flex

gochicus committed:
refactor(list): package description rewritten

ISSUES CLOSED: #81
